src/Handlers/BookingsStateStatusTransitionsHandler.php

```diff
@@ -0,0 +1,145 @@
+use Dhii\Invocation\InvocableInterface;
+use Dhii\Util\Normalization\NormalizeArrayCapableTrait;
+use Psr\EventManager\EventInterface;
+use stdClass;
+use Traversable;
+
+/**
+ * Handler for adding transitions information to state on bookings page.
+ * It will add map of available statuses transitions and transitions translations.
+ *
+ * @since [*next-version*]
+ */
+class BookingsStateStatusTransitionsHandler implements InvocableInterface
+{
+    /* @since [*next-version*] */
+    use NormalizeArrayCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateInvalidArgumentExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use StringTranslatingTrait;
+
+    /**
+     * Map of transitions that should be applied when changing status.
+     *
+     * @since [*next-version*]
+     *
+     * @var array|Traversable|stdClass
+     */
+    protected $statusTransitions;
+
+    /**
+     * List of labels for transitions.
+     *
+     * @since [*next-version*]
+     *
+     * @var array|Traversable|stdClass
+     */
+    protected $transitionsLabels;
+
+    /**
+     * StatusesHandler constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|Traversable|stdClass $statusTransitions Map of transitions for changing statuses.
+     * @param array|Traversable|stdClass $transitionsLabels Map of transitions for changing statuses.
+     */
+    public function __construct(
+        $statusTransitions,
+        $transitionsLabels
+    ) {
+        $this->statusTransitions = $statusTransitions;
+        $this->transitionsLabels = $transitionsLabels;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke()
+    {
+        /* @var $event EventInterface */
+        $event = func_get_arg(0);
+
+        if (!($event instanceof EventInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an event instance'), null, null, $event
+            );
+        }
+
+        $event->setParams(array_merge(
+            $event->getParams(),
+            $this->_getTransitionParams()
+        ));
+    }
+
+    /**
+     * Get transitions related parameters to attach on event.
+     *
+     * @since [*next-version*]
+     *
+     * @return array
+     */
+    protected function _getTransitionParams()
+    {
+        return [
+            /*
+             * Transitions for changing booking status
+             */
+            'statusTransitions' => $this->_prepareStatusTransitions($this->statusTransitions),
+
+            /*
+             * Map of transitions keys to transition labels
+             */
+            'transitionsLabels' => $this->_prepareTransitionsLabels($this->transitionsLabels),
+        ];
+    }
+
+    /**
+     * Get map of status transitions.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|Traversable|stdClass $statusTransitions Map of status transitions.
```

Comment:
Although the status transitions accepted in the constructor may become a map, here accepting a `Traversable` is still OK, because you are working with it as a list.